File: leetcode/march2024/24_find_duplicate_number.cpp
// CODE

class Solution {
    public:
    int findDuplicate(vector<int>& nums) {
        int res = 0, n = nums.size();
        for(int i = 0; i < n; i++) {
            nums[nums[i] % n] += n;
        }
        for(int i = 0; i < n; i++) {
            if(nums[i] / n > 1) res = i;
            nums[i] %= n;
        }
        return res;
    }
};

File: leetcode/march2024/python/24_find_duplicate_number.py
# CODE

class Solution:
    def findDuplicate(self, nums: List[int]) -> int:
        n = len(nums)
        res = 0
        for i in range(0, n):
            nums[nums[i] % n] += n
        for i in range(0, n):
            if nums[i] // n > 1: 
                res = i
            nums[i] %= n
        return res   